File: 128/Solution_128.java

public class Solution_128 {
    public static void main(String[] args){

    }
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) return 0;

        Arrays.sort(nums);
        int currentSeq = 1;
        int longestSeq = 1;

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] != nums[i-1]) {  // Skip duplicates
                if (nums[i] == nums[i-1] + 1) {
                    currentSeq++;
                } else {
                    longestSeq = Math.max(longestSeq, currentSeq);
                    currentSeq = 1;
                }
            }
        }

        return Math.max(longestSeq, currentSeq);
    }
    public int longestConsecutive(int[] nums){
        Set<Integer> numSet = new HashSet<>();
        for (int num : nums){
            numSet.add(num);
        }

        int longestSeq = 0;

        for (int num : numSet){
            if (!numSet.contains(num-1)){
                int cur = num;
                int curSeq = 1;
            }

            while (numSet.contains(cur+1)){
                cur++;
                curSeq++;
            }

            longestSeq = Math.max(longestSeq,curSeq);
        }
        
        return longestSeq;

    }
}

File: 128/Solution_128.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int ans = 0;
        unordered_set<int> st(nums.begin(), nums.end()); // 把 nums 转成哈希集合
        for (int x : st) { // 遍历哈希集合
            if (st.contains(x - 1)) {
                continue;
            }
            // x 是序列的起点
            int y = x + 1;
            while (st.contains(y)) { // 不断查找下一个数是否在哈希集合中
                y++;
            }
            // 循环结束后，y-1 是最后一个在哈希集合中的数
            ans = max(ans, y - x); // 从 x 到 y-1 一共 y-x 个数
        }
        return ans;
    }
};

